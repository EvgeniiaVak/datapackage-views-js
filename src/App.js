import React from "react"
import "./App.css"
import Table from "./Table.js"
import {handsOnTableToHandsOnTable} from 'datapackage-render'

function App(props) {
  let _data = props.view ? props.view.resources[0].data : []
  let view = props.view || {} // default to single table view
  if (view.resources) view.resources[0]._values =  _data
  let {data, ...options} = handsOnTableToHandsOnTable(view)
  let renderedView = <p>Data view unavailable</p>

  if (view.specType === "table" || !view.specType) {
    renderedView = <Table data={data} options={options} />
  }

  return (
    <div className="App">
      <div className="container m-24">{renderedView}</div>
    </div>
  )
}

export default App
