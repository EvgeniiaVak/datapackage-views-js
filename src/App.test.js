import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import App from './App';


const view = {
  'id': 1,
  'title': 'GDP',
  'resources': [
    'gdp'
  ],
  'specType': 'table'
}

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App view={view} />, div)
  ReactDOM.unmountComponentAtNode(div)
});

it('renders loading message', () => {
  const { getByText } = render(<App view={view} />)
  expect(getByText('Data view is loading')).toBeInTheDocument()
});

it('renders error message when data is unavailable', () => {
  const errorView = JSON.parse(JSON.stringify(view))
  errorView.resources[0] = {
    name: 'gdp',
    unavailable: true
  }
  const { getByText } = render(<App view={errorView} />)
  expect(getByText('Data view unavailable.')).toBeInTheDocument()
})

it('renders a preview table when data is loaded', () => {
  const compiledView = JSON.parse(JSON.stringify(view))
  compiledView.resources[0] = {
    name: 'gdp',
    data: [
      {a:1, b:2},
      {a:3, b:4}
    ]
  }
  const { container } = render(<App view={compiledView} />)
  expect(container.querySelector('table.htCore')).toMatchSnapshot()
})

it('renders a preview table with custom headers', () => {
  const compiledView = JSON.parse(JSON.stringify(view))
  compiledView.resources[0] = {
    name: 'gdp',
    data: [
      {a:1, b:2},
      {a:3, b:4}
    ],
    schema: {
      fields: [
        {name: 'a', title: 'custom-a'},
        {name: 'b', title: 'custom-b'}
      ]
    }
  }
  const { container } = render(<App view={compiledView} />)
  expect(container.querySelector('table.htCore')).toMatchSnapshot()
})

it('renders a Map for geojson resources', () => {
  const compiledView = JSON.parse(JSON.stringify(view))
  compiledView.specType = 'map'
  compiledView.resources[0] = {
    name: 'map',
    data: {
      "type": "Feature",
      "geometry": {
        "type": "Point",
        "coordinates": [
          125.6,
          10.1
        ]
      },
      "properties": {
        "name": "Dinagat Islands"
      }
    }
  }
  const { container } = render(<App view={compiledView} />)
  expect(container.firstChild).toMatchSnapshot()
})

it('renders a Document for PDF resources', () => {
  const compiledView = JSON.parse(JSON.stringify(view))
  compiledView.specType = 'document'
  compiledView.resources[0] = {
    name: 'document',
    format: 'pdf',
    path: 'some-path-to-pdf'
  }
  const { container } = render(<App view={compiledView} />)
  expect(container.firstChild).toMatchSnapshot()
})
