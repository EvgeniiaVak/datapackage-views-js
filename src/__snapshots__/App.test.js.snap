// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a preview table when data is loaded 1`] = `
<table
  class="htCore"
>
  <colgroup>
    <col
      style="width: 47px;"
    />
  </colgroup>
  <thead>
    <tr>
      <th
        class=""
        title="a"
      >
        <div
          class="relative"
        >
          <span
            class="colHeader columnSorting sortAction"
          >
            a
          </span>
        </div>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        class="htDimmed"
      >
        1
      </td>
    </tr>
    <tr>
      <td
        class="htDimmed"
      >
        3
      </td>
    </tr>
  </tbody>
</table>
`;
